src/interface/variable.hpp

```diff
@@ -118,18 +118,24 @@ class CellVariable {
 
   inline bool IsSet(const MetadataFlag bit) const { return m_.IsSet(bit); }
 
-  ParArrayND<T> data;
-  ParArrayND<T> flux[4];  // used for boundary calculation
-  ParArrayND<T> coarse_s; // used for sending coarse boundary calculation
+  ParArrayND<T, VariableState> data;
+  ParArrayND<T, VariableState> flux[4];  // used for boundary calculation
+  ParArrayND<T, VariableState> coarse_s; // used for sending coarse boundary calculation
 
   int dealloc_count = 0;
 
+  int GetAllocationStatus() {
+    if (!is_allocated_) return 0;
+    return num_alloc_;
+  }
+
  private:
   // allocate data, fluxes, and boundary variable
-  void Allocate(std::weak_ptr<MeshBlock> wpmb);
+  void Allocate(std::weak_ptr<MeshBlock> wpmb, bool flag_uninitialized = false);
+  int num_alloc_ = 0;
```

Comment:
It should increase by one every time the field is allocated, so after allocation->deallocation->allocation it should be increased by two. This gives a way to check if the data that is pointed to in a pack is stale or not (since just checking the allocation status of fields can resulting in thinking your pack is not stale when it is).